src/pages/RepoPage/FailedTestsTab/FailedTestsPage/hooks/useTestResultsTestSuites/useTestResultsTestSuites.tsx

```diff
@@ -0,0 +1,97 @@
+import { useQuery } from '@tanstack/react-query'
+import { useParams } from 'react-router-dom'
+import { z } from 'zod'
+
+import { RepoNotFoundErrorSchema } from 'services/repo'
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+const TestResultsTestSuitesSchema = z.object({
+  owner: z
+    .object({
+      repository: z.discriminatedUnion('__typename', [
+        z.object({
+          __typename: z.literal('Repository'),
+          testAnalytics: z
+            .object({
+              testSuites: z.array(z.string()),
+            })
+            .nullable(),
+        }),
+        RepoNotFoundErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `
+  query GetTestResultsTestSuites(
+    $owner: String!
+    $repo: String!
+    $term: String
+  ) {
+    owner(username: $owner) {
+      repository: repository(name: $repo) {
+        __typename
+        ... on Repository {
+            testAnalytics {
+             testSuites(term: $term)
+          }
+        }
+        ... on NotFoundError {
+          message
+        }
+      }
+    }
+  }
+  `
+
+interface URLParams {
+  provider: string
+  owner: string
+  repo: string
+}
+
+export const useTestResultsTestSuites = ({ term }: { term?: string }) => {
+  const { provider, owner, repo } = useParams<URLParams>()
+
+  return useQuery({
+    queryKey: ['GetTestResultsTestSuites', provider, owner, repo, term],
+    queryFn: ({ signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: {
+          provider,
+          owner,
+          repo,
+          term,
+        },
+      }).then((res) => {
+        const parsedData = TestResultsTestSuitesSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
```

Comment:
Not blocking or needed now - with the new pattern per [here](https://github.com/codecov/gazebo/pull/3436) if you use the helper `rejectNetworkError` it will send to sentry which is handy

Reply:
PR up here! https://github.com/codecov/gazebo/pull/3484